### DISCUSSION TITLE
Data augmentation

### DISCUSSION THREAD

**Discussion by robsver** [Project-MONAI/MONAI#6021] [General]
Hi all,
I am wondering how to run image augmentation during a training.
Let me explain it more in details:

I load the images (3D) by using ImageDataset and DataLoader from MONAI infrastructure.

Starting from each single patient, I want to obtain 10 more different new images by applying some transforms (that I define in the code) in a random order and by using a random combination of parameters for each transform.

In addition, I don't want to feed the model with a series of images coming from the same patient even if augmented (e.g. img1, img1_augmented_version1, img1_augmented_version2, img2, img2_augmented_version1, img2_augmented_version2), but I want to show them into shuffled order (e.g. img1, img2_augmented_version3, img1_augmented_version6, img5, etc...).

Do you have any suggestions?

Thank you very much in advance.

Best,
Roberto

**Comment by KumoLiu** [comment]
Hi @robsver, thanks for you interest here. In some random transforms, we have `num_samples` which can help you to take several samples from one image. But we don't have a method to shuffle the order of these samples randomly. You could set `shuffle=True` to shuffle the order of cases in training step.
https://github.com/Project-MONAI/MONAI/blob/bf55f22c461bac286e3078a242d9d0b90a43ba00/monai/transforms/croppad/array.py#L993
```
train_loader = DataLoader(train_ds, batch_size=1, shuffle=True, num_workers=4)
```
Hope it can help you, thanks!

**Reply by robsver** [reply]
Hi @KumoLiu thanks for answering but I have to find another way.

Do you know if, in the future, will be added this kind of feature that I'm looking for?

Thanks again.

**Reply by wyli** [reply]
there are some related patch-based analysis tutorials about the topics you mentioned @robsver, please have a look:
- https://github.com/Project-MONAI/tutorials/blob/main/modules/2d_slices_from_3d_training.ipynb
- https://github.com/Project-MONAI/tutorials/tree/main/modules/generate_random_permutations
- https://github.com/Project-MONAI/tutorials/blob/main/modules/2d_slices_from_3d_sampling.ipynb
these are written in terms of sampling 2d from 3d, but it can generate 3d samples as well.

**Reply by robsver** [reply]
Hi @wyli I really appreciate it, thank you!
